samples/crypto/tls/src/tls_functions_server.c

```diff
@@ -0,0 +1,173 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tls_functions);
+
+#include <nrf.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <net/socket.h>
+#include <net/net_core.h>
+#include <net/tls_credentials.h>
+#include <linker/sections.h>
+
+#include "certificate.h"
+#include "psa_tls_functions.h"
+#include "psa_tls_credentials.h"
+
+
+char m_recv_buffer[RECV_BUFFER_SIZE]; /* Buffer for storing payload */
+
+/* Define thread parameters for TLS Server. */
+static int setup_tls_server_socket(void);
+static void process_tls_server(void);
+K_THREAD_DEFINE(tls_server_thread_id, THREAD_STACK_SIZE, process_tls_server,
+		NULL, NULL, NULL, THREAD_PRIORITY, 0, -1);
+
+
+void tls_start_thread(void)
+{
+	k_thread_start(tls_server_thread_id);
+}
+
+
+/** @brief Function for sending all data in a given buffer to a connected
+ * socket.
+ *
+ * @param   sock        The socket identifier to send data to.
+ * @param   buf         Pointer to the data buffer.
+ * @param   len         Number of bytes to send from buf.
+ *
+ * @returns 0 on success.
+ */
+static ssize_t send_buffer(int sock, const uint8_t *buf, size_t len)
+{
+	ssize_t sent_len;
+
+	while (len > 0) {
+		sent_len = send(sock, buf, len, 0);
+		if (sent_len < 0) {
+			return sent_len;
+		}
+
+		buf = (const uint8_t *)buf + sent_len;
+		len -= sent_len;
+	}
+
+	return 0;
+}
+
+
+/** @brief Function for setting up the server socket to use for TLS handshake.
+ *
+ * @returns The socket identifier.
+ */
+static int setup_tls_server_socket(void)
+{
+	int err;
+	int sock;
+	struct sockaddr_in my_addr;
+
+	/* List of security tags to register. */
+	sec_tag_t sec_tag_list[] = {
+		SERVER_CERTIFICATE_TAG,
+		#if defined(CONFIG_MBEDTLS_KEY_EXCHANGE_PSK_ENABLED)
+		PSK_TAG,
+		#endif
+	};
+
+	memset(&my_addr, 0, sizeof(my_addr));
+	my_addr.sin_family = AF_INET;
+	my_addr.sin_port = htons(SERVER_PORT);
+	sock = socket(my_addr.sin_family, SOCK_STREAM, IPPROTO_TLS_1_2);
+
+	err = setsockopt(sock, SOL_TLS, TLS_SEC_TAG_LIST, sec_tag_list,
+			 sizeof(sec_tag_list));
+	if (err < 0) {
+		LOG_ERR("Failed to set TLS security TAG list. Err: %d", errno);
+		return -errno;
+	}
+
+	err = bind(sock, (struct sockaddr *)&my_addr, sizeof(my_addr));
+	if (err < 0) {
+		LOG_ERR("Failed to bind TLS socket. Err: %d", errno);
+		return -errno;
+	}
+
+	err = listen(sock, MAX_CLIENT_QUEUE);
+	if (err < 0) {
+		LOG_ERR("Failed to listen on TLS socket. Err: %d", errno);
+		return -errno;
+	}
+
+	return sock;
+}
+
+/** @brief Thread function that handles all the steps of the TLS handshake for
+ *  a server. Loops forever and accepts connections on the given port number.
+ *
+ */
+static void process_tls_server(void)
+{
+	int err;
+	int sock;
+	int client;
+	int received;
+	struct sockaddr_in client_addr;
+	socklen_t client_addr_len = sizeof(client_addr);
+
+	while (true) {
+		sock = setup_tls_server_socket();
+		if (sock < 0) {
+			LOG_INF("Retrying to aquiring socket");
```

Comment:
```suggestion
			LOG_INF("Retrying to create a socket");
```

Generally, the same comments as in the "client" part also apply here.